rest-api/dao/physical_measurements_dao.py

```diff
@@ -291,32 +291,49 @@ def _update_participant_summary(self, session, obj):
-      if not self.has_uncancelled_pm(session, participant):
-        # update corresponding participant summary physicalMeasurementsStatus
-        participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
-        participant_summary.physicalMeasurementsTime = None
+    # These fields set on measurement that is cancelled and doesn't have a previous good measurement
+    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and not \
+       self.has_uncancelled_pm(session, participant):
 
-    if participant_summary.physicalMeasurementsStatus != PhysicalMeasurementsStatus.CANCELLED or \
-       obj.status != PhysicalMeasurementsStatus.CANCELLED:
-      # if a PM was restored, it is complete again.
+      participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
+      participant_summary.physicalMeasurementsTime = None
+
+    # These fields set on any measurement not cancelled
+    elif obj.status != PhysicalMeasurementsStatus.CANCELLED:
+      # new PM or if a PM was restored, it is complete again.
       participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.COMPLETED
+      participant_summary.physicalMeasurementsTime = obj.created
+      participant_summary.physicalMeasurementsFinalizedTime = obj.finalized
+      participant_summary.physicalMeasurementsCreatedSiteId = obj.createdSiteId
+      participant_summary.physicalMeasurementsFinalizedSiteId = obj.finalizedSiteId
+
+    elif obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and \
+       self.has_uncancelled_pm(session, participant):
+
+      get_latest_pm = self.get_latest_pm(session, participant)
+      participant_summary.physicalMeasurementsFinalizedTime = get_latest_pm.finalized
+      participant_summary.physicalMeasurementsTime = get_latest_pm.created
+      participant_summary.physicalMeasurementsCreatedSiteId = get_latest_pm.createdSiteId
+      participant_summary.physicalMeasurementsFinalizedSiteId = get_latest_pm.finalizedSiteId
 
     participant_summary_dao.update_enrollment_status(participant_summary)
     session.merge(participant_summary)
 
     return participant_summary
 
+  def get_latest_pm(self, session, participant):
+    return session.query(PhysicalMeasurements).filter_by(
+      participantId=participant.participantId).filter(PhysicalMeasurements.finalized !=
+                                                      None).order_by(
+                                                      PhysicalMeasurements.finalized.desc()).first()
+
   def has_uncancelled_pm(self, session, participant):
     """return True if participant has at least one physical measurement that is not cancelled"""
     query = session.query(PhysicalMeasurements.status).filter_by(
-      participantId=participant.participantId).all()
+      participantId=participant.participantId).filter(PhysicalMeasurements.finalized !=
```

Comment:
I don't see why this change was necessary - are there going to be uncancelled PMs which don't have a finalized time?

Reply:
There can be. Not sent from HPRO but if another order was cancelled and finalized was set to None it could be returned here.